YDA-5789: limit max number of client connections

lwesterhof opened the pull request:
https://github.com/UtrechtUniversity/yoda/pull/445

What is an acceptable default max user connections?

lwesterhof committed:
YDA-5789: limit max number of client connections

claravox committed:
YDA-5789: do not limit anonymous connections

claravox commented on util/config.py:
```diff
@@ -145,6 +145,7 @@ def __repr__(self):
                 vault_copy_backoff_time=300,
                 vault_copy_max_retries=5,
                 vault_copy_multithread_enabled=True,
+                user_max_connections=3,
```
```suggestion
                user_max_connections=4,
```

lwesterhof committed:
YDA-5789: max number of client connections to 4

Co-authored-by: claravox <claravox@users.noreply.github.com>

claravox reviewed (approved):
LGTM

stsnel commented on util/user.py:
```diff
@@ -104,3 +105,16 @@ def name_from_id(ctx, user_id):
+        username = session_vars.get_map(ctx.rei)['client_user']['user_name']
+        connections = ips.count(username)
+    except Exception:
+        return 0
```
I think it would be good to also log the exception here, since this is an unexpected situation.

stsnel committed:
user.number_of_connections: log exceptions

stsnel committed:
user.number_of_connections: use ips without -a

Check the number of connections only locally (no -a option),
because checking all servers can take a lot of time in a production
setting.

stsnel committed:
YDA-5789: update max_connections_exceeded policy

- Only rods user is exempt, rather than all rodsadmin users.
- Add parameter for enabling / disabling the max_connections check
- Update other config parameter name for this policy

stsnel committed:
check_max_connections policy: fix threshold

The number of connections that is measured for a user includes the
present one, so we should still accept a connection if the current
number of connections is equal to the maximum number.

stsnel reviewed (commented):
LGTM :+1:

lwesterhof reviewed (commented):
LGTM
